### ISSUE TITLE
Critical bug - Databasepassword is visible when MySQL not reacheable

### ISSUE THREAD

**Issue by Th3f** [anchorcms/anchor-cms#814]
When the MySQL-Server isn't reacheable, one's able to read out the password and databasename: 
![scr22-26-09](https://cloud.githubusercontent.com/assets/8012652/6001512/08f41d3c-aaeb-11e4-99e8-1af9c4931747.png)
(You can see it at the and of # 1 in trace)

**Comment by TimohStudios** [comment]
That's both really bad and really weird. @CraigChilds94 or @kieronwilson, any chance you can fix this any time soon?

**Comment by daviddarnes** [comment]
I think error reporting had changed in the most recent build?

**Comment by sambuev** [comment]
no way! You have to report to every users on the main page.!

**Comment by pinodex** [comment]
There's a code in [rwarasaurus/nano](https://github.com/rwarasaurus/nano/blob/master/system/database/connector.php#L56) that might help.

![screenshot from 2015-02-25 14 26 40](https://cloud.githubusercontent.com/assets/6258767/6366024/7a293110-bcfa-11e4-8656-cf1c64f5434e.png)

I haven't tested it yet.

**Comment by NinjaSentry** [comment]
It doesn't seem like a critical bug. When you throw an exception, you also need to catch it.

You could use an Exception Handler to deal with any uncaught exceptions :
http://php.net/manual/en/function.set-exception-handler.php

Also use your environment to help control when errors are displayed throughout your app. If it's on a live production server, set display errors to false.

**Comment by CraigChilds94** [comment]
I've added a system which catches exceptions when Anchor trys to start, hopefully this won't be an issue now. But for dev purposes you will still get exceptions display if you're in `dev` mode etc...

Also all errors are now logged in `/anchor/errors.log`, i've put them on a line/line log using json to format the data. Check out `0.9-dev` branch to see all the changes that I've made. I'm aiming to get another tag out soon.
